docs/topics/kotlin-daemon.md

```diff
@@ -0,0 +1,106 @@
+[//]: # (title: Kotlin daemon)
+
+The Kotlin daemon is a background process that build systems can use to improve Kotlin compilation performance.
+The Kotlin daemon speeds up build times by keeping the compiler and its environment ready to compile. This approach 
+avoids the overhead of starting a new Java Virtual Machine (JVM) instance and reinitializing the compiler for every 
+compilation, leading to reduced build times for incremental builds or frequent small changes.
+
+Some build systems, like [Gradle](https://docs.gradle.org/current/userguide/gradle_daemon.html) and [Maven](https://maven.apache.org/tools/mvnd.html),
+have their own daemons that help reduce startup costs. However, they miss the added benefit that the Kotlin daemon
+completely isolates the build system process from the compiler. This separation is useful in dynamic environments where
+system settings might change at runtime.
+
+Although the Kotlin daemon has no direct user-facing interface, you can use it through build systems or the [build tools API](build-tools-api.md).
+
+## Kotlin daemon configuration
+
+There are ways to configure some settings for the Kotlin daemon for Gradle or Maven.
+
+### Memory management
+
+The Kotlin daemon is a separate process that has its own memory space, isolated from the client.
+By default, the Kotlin daemon tries to inherit the heap size (`-Xmx`) of the launching JVM process.
+
+To configure specific memory limits, like `-Xmx` and `-XX:MaxMetaspaceSize`, use the following property:
+
+<tabs group="build-system">
+<tab title="Gradle" group-key="gradle">
+
+```kotlin
+kotlin.daemon.jvmargs=-Xmx1500m
+```
+
+For more information, see [`kotlin.daemon.jvmargs` property](gradle-compilation-and-caches.md#kotlin-daemon-jvmargs-property).
+
+</tab>
+<tab title="Maven" group-key="maven">
+
+```xml
+<kotlin.compiler.daemon.jvmArgs>-Xmx1500m</kotlin.compiler.daemon.jvmArgs>
+```
+
+</tab>
+</tabs>
+
+### Lifetime
+
+There are two common lifetime strategies for the Kotlin daemon:
+
+* **Attached daemon**: Used when the client (like a build system) is long-running. The daemon shuts down shortly after the
+client process shuts down or the daemon hasn't been used for a while.
+* **Detached daemon**: Used when the client is short-lived (like command-line compilation). The daemon stays alive longer
+to await potential follow-up requests.
```

Comment:
After this, I expected "attached daemon / detached daemon" or "lifetime strategy" to be specific programmatic categories, which are to be used. But as I understand the user first needs to implement a strategy, then use it. So maybe it would be clearer like this:

```
There are two common strategies that you can implement for the Kotlin daemon:

* Attached daemon: Shut down the daemon shortly after <...>. Use when the client is long-running.
* Detached daemon: Keep the daemon alive longer <...>. Use when the client is short-lived.
```